internal/errors/errors.go

```diff
@@ -12,7 +12,7 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-package cmd
+package errors
```

Comment:
`NewValidationError` makes feel wtf does "validation" mean, and the reality is that it is basically just aliasing `fmt.Errorf`. Do we really need this package now?

Reply:
validate is pre-flight, basically something that happens before download. I would like to represent it because we do actually use this type. It is used to print a usage hint if we know it is argument related.. any ideas?

Reply:
ps this might help https://github.com/tetratelabs/getenvoy/blob/master/main.go#L48

in general we still have way too many packages, ex "cmd" vs "binary" etc and maybe this one stands out because I used it before. Once the main entanglement caused by the object model is done, I'd be happy to reduce this and more.

Ex. you'll notice the "cmd" thing makes the hierarchy more bizarre also
```
internal
internal/globals
internal/cmd
internal/transport
internal/test
internal/test/morerequire
internal/test/manifest
internal/manifest
internal/version
internal/binary
internal/binary/envoytest
internal/binary/envoy
internal/binary/envoy/debug
internal/errors
internal/tar
```

Reply:
https://github.com/tetratelabs/getenvoy/issues/235 but we can still discuss "errors" here. Hopefully the additional doc I put on that type makes sense. I don't know a better word for validate at the moment, so if you can suggest one, I'm all ears also!

Reply:
I went with up-front validation for now. There's some more refactoring we can do later wrt where validation occurs, but this should get the distraction out of this PR!